src/PlatformServices/Utilities/AssemblyUtility.cs

```diff
@@ -81,15 +104,67 @@ internal bool IsAssemblyExtension(string extensionWithLeadingDot)
             return false;
         }
 
+
+        /// <summary>
+        /// Verifies if the assembly provided is referenced by the source.
+        /// </summary>
+        /// <param name="assemblyName"> The assembly name. </param>
+        /// <param name="source"> The source. </param>
+        /// <returns> True if the assembly is referenced. </returns>
+        public static bool IsAssemblyReferenced(AssemblyName assemblyName, string source)
+        {
+#if NETSTANDARD1_4
+#elif NETFRAMEWORK
+            // This loads the dll in a different app domain.
+            // If no reference to UTF don't run discovery. Take conservative approach. If not able to find proceed with discovery.
+            var utfReference = AssemblyHelper.DoesReferencesAssembly(source, assemblyName) ?? true;
+            if (!utfReference)
```

Comment:
Looks like you could simply return `utReference` and move the final `return true` in the pragma.